packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.test.tsx

```diff
@@ -40,6 +40,20 @@ describe('DatePicker', () => {
     expect(wrapper.state('isDatePickerShown')).toBe(false);
   });
 
+  it('should call onSelectDate even when required input is empty when allowTextInput is true', done => {
+    const onSelectDate = (date?: Date) => {
+      if (!date) {
+        done();
+      }
+    };
+    const datePicker = mount(<DatePickerBase isRequired={true} allowTextInput={true} onSelectDate={onSelectDate} />);
+    const textField = datePicker.find('input');
+    expect(textField).toBeDefined();
+    textField.simulate('change', { target: { value: 'Jan 1 2030' } }).simulate('blur');
+    textField.simulate('change', { target: { value: '' } }).simulate('blur');
+    datePicker.unmount();
```

Comment:
I would recommend modifying this test's success criteria to check that a mocked `onSelectDate` is called X number of times. You can verify parameters also, but I think the following should suffice:

```tsx
it('should call onSelectDate even when required input is empty when allowTextInput is true', done => {
  const onSelectDate = jest.fn();
  const datePicker = mount(<DatePickerBase isRequired={true} allowTextInput={true} onSelectDate={onSelectDate} />);
  const textField = datePicker.find('input');

  expect(textField).toBeDefined();

  textField.simulate('change', { target: { value: 'Jan 1 2030' } }).simulate('blur');
  textField.simulate('change', { target: { value: '' } }).simulate('blur');

  expect(onSelectDate).toHaveBeenCalledTimes(2);
  
  datePicker.unmount();
});
```

What do you think @nickdurak? My assumption here is that the `onSelectDate` should be called `twice`. Once with a text value and the second time with an empty value.

Reply:
Currently, the test is requiring an empty value be passed in order for the test callback to be completed. This could cause side effects, hence my recommendation above.

Thank you for adding the test coverage!

Reply:
sounds good, just updated

Reply:
@KevinTCoughlin can you review the update?

Reply:
👍 thank you for updating.